frontend/public/components/default-resource.tsx

```diff
@@ -148,9 +219,28 @@ export const DefaultList: React.FC<TableProps & { kinds: string[] }> = (props) =
   const { t } = useTranslation();
 
   const { kinds } = props;
+  const [model] = useK8sModel(kinds[0]);
+  const additionalPrinterColumns = getMaxAdditionalPrinterColumns(
+    useCRDAdditionalPrinterColumns(model),
+  );
 
   const TableHeader = () => {
-    return [
+    const additionalPrinterColumnsHeaders = additionalPrinterColumns.map((col) => {
+      const path = col.jsonPath;
+      const pathHasSpecialCharacter = checkPathHasSpecialCharacter(path);
+
+      return {
+        title: col.name,
+        sortField: pathHasSpecialCharacter ? undefined : path.replace(/^\./, ''),
```

Comment:
I think we can make this work with any JSON path expression by using a custom sort function, but I'd have to look at how that works in our table component.